applications/StructuralMechanicsApplication/custom_processes/set_automated_initial_variable_process.cpp

```diff
@@ -0,0 +1,124 @@
+    
+    array_1d<double, 3> hole_generatrix_point = mThisParameters["hole_generatrix_point"].GetVector();  
+
+    array_1d<double, 3> normalized_generatrix_vector;   
+    normalized_generatrix_vector[0] = (hole_generatrix_axis[0] * hole_generatrix_axis[0]) / std::sqrt(hole_generatrix_axis[0] * hole_generatrix_axis[0] + hole_generatrix_axis[1] * hole_generatrix_axis[1] + hole_generatrix_axis[2] * hole_generatrix_axis[2]);
+    normalized_generatrix_vector[1] = (hole_generatrix_axis[1] * hole_generatrix_axis[1]) / std::sqrt(hole_generatrix_axis[0] * hole_generatrix_axis[0] + hole_generatrix_axis[1] * hole_generatrix_axis[1] + hole_generatrix_axis[2] * hole_generatrix_axis[2]);
+    normalized_generatrix_vector[2] = (hole_generatrix_axis[2] * hole_generatrix_axis[2]) / std::sqrt(hole_generatrix_axis[0] * hole_generatrix_axis[0] + hole_generatrix_axis[1] * hole_generatrix_axis[1] + hole_generatrix_axis[2] * hole_generatrix_axis[2]);
+
+    const double hole_radius_offset = mThisParameters["hole_radius_offset"].GetDouble();
+    const double tol = 1.0e-6;
+
+    block_for_each(mrThisModelPart.Elements(), [&](Element &rElement) {
+
+        array_1d<double, 3> element_centroid = rElement.GetGeometry().Center();
+        
+        array_1d<double, 3> relative_position_vector;
+        relative_position_vector[0] = element_centroid[0] - hole_generatrix_point[0];
+        relative_position_vector[1] = element_centroid[1] - hole_generatrix_point[1];
+        relative_position_vector[2] = element_centroid[2] - hole_generatrix_point[2];
+
+        double vector_scaler = relative_position_vector[0] * normalized_generatrix_vector[0] + relative_position_vector[1] * normalized_generatrix_vector[1] + relative_position_vector[2] * normalized_generatrix_vector[2];
+        
+        array_1d<double, 3> intersection_point = hole_generatrix_point + vector_scaler * normalized_generatrix_vector;
+
+        array_1d<double, 3> radial_position_vector = element_centroid - intersection_point;
+
+        double centroid_relative_distance = std::sqrt(radial_position_vector[0] * radial_position_vector[0] + radial_position_vector[1] * radial_position_vector[1] + radial_position_vector[2] * radial_position_vector[2]) - hole_radius_offset;
+
+        if (centroid_relative_distance < 0.0 && abs(centroid_relative_distance) <= tol) {
+            centroid_relative_distance = 0.0;
+        }
+        else if(centroid_relative_distance < 0.0 && abs(centroid_relative_distance) >= tol){
+            int ElemId = rElement.Id();
+            KRATOS_ERROR << "Thickness of element " << ElemId << " is too small." << std::endl;
+        }
+
+        int TableFirstId = mThisParameters["initial_variable_table"]["table_id"].GetInt() - mrThisModelPart.Tables().size()+1;
+        
+        int count = 0;
+        array_1d<double, 6> initial_variable_vector = ZeroVector(); 
```

Comment:
array_1d<double, 6> initial_variable_vector;
noalias(initial_variable_vector) = ZeroVector(6);

which means is only for 3d right?

Reply:
Yes. It is designed for the 3d case.